### PR TITLE
feat(drag-drop): adds light/dark mode colors

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Colors updated for the following components & stories using new v9 color palette:

- Draggable
    - Draggable.Grip
- DraggableList.DropIndicator
- Dropzone

## Checklist

- [x] :ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [x] :wheelchair: tested for WCAG 2.1 AA accessibility compliance
- ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

**Comment by geotrev** [comment]
Waiting for new disabled color variables from #1806

File: packages/drag-drop/src/styled/dropzone/StyledDropzone.ts [hunk 1]
```diff
@@ -24,25 +24,51 @@ export interface IStyledDropzoneProps extends ThemeProps<DefaultTheme> {
...
+  let borderColor = getColor({ variable: `border.emphasis`, theme });
+  let color = getColor({ variable: `foreground.subtle`, theme });
 
   if (isDisabled) {
-    backgroundColor = getColorV8('neutralHue', 200, theme) as string;
-    borderColor = getColorV8('neutralHue', 300, theme);
-    color = getColorV8('neutralHue', 400, theme);
+    backgroundColor = getColor({
+      variable: 'background.disabled',
+      transparency: theme.opacity[100],
```
Comment (jzempel) [inline packages/drag-drop/src/styled/dropzone/StyledDropzone.ts:45]: Can remove `transparency` – already built-in

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/drag-drop/src/styled/draggable-list/StyledDropIndicator.ts [hunk 2]
```diff
@@ -16,15 +16,14 @@ export interface IStyledDropIndicatorProps extends ThemeProps<DefaultTheme> {
 
 const colorStyles = (props: IStyledDropIndicatorProps) => {
   const { theme } = props;
-  const backgroundColor = getColorV8('primaryHue', 600, theme);
-  const color = getColorV8('primaryHue', 600, theme);
+  const color = getColor({ variable: 'foreground.primary', theme });
```
[the remark this answers is not in the document]
Comment (geotrev) [reply packages/drag-drop/src/styled/draggable-list/StyledDropIndicator.ts:19]: That's fair. I was teetering between fg and border, anyway. :)

File: packages/drag-drop/demo/~patterns/stories/components.tsx [hunk 3]
```diff
@@ -216,6 +218,10 @@ export const DraggableListItem = ({
   );
 };
 
+const StyledTitle = styled.strong`
+  color: ${p => getColor({ variable: 'foreground.default', theme: p.theme })};
```
[the remark this answers is not in the document]
Comment (geotrev) [reply packages/drag-drop/demo/~patterns/stories/components.tsx:222]: I also just realized I can convert this to a more semantically correct `h2`.

[the remark this answers is not in the document]
Comment (jzempel) [reply packages/drag-drop/demo/~patterns/stories/components.tsx:223]: [nit] the `h2` still has the bold font weight problem. Would it be better to use Garden's `<LG tag="h2">` typography component? We just don't want to confuse any consumers that view the pattern.

[the remark this answers is not in the document]
Comment (geotrev) [reply packages/drag-drop/demo/~patterns/stories/components.tsx:223]: Yeah that makes sense. I'll update.

**Review by jzempel — APPROVED** [review]

**Pull request merged**